accepted/2021/statics-in-interfaces/README.md

```diff
@@ -6,8 +6,1267 @@
+    {
+        // These do not take NumberStyles as not all parseable types are numbers (such as Guid)
+        // Instead, INumber<TSelf> exposes additional parse methods that do take NumberStyles
+
+        static abstract TSelf Parse(string s, IFormatProvider? provider);
+
+        static abstract bool TryParse(string s, IFormatProvider? provider, out TSelf result);
+    }
+
+    public interface ISpanParseable<TSelf> : IParseable<TSelf>
+        where TSelf : ISpanParseable<TSelf>
+    {
+        // This inherits from IParseable, much as ISpanFormattable is planned to inherit from IFormattable
+
+        static abstract TSelf Parse(ReadOnlySpan<char> s, IFormatProvider? provider);
+
+        static abstract bool TryParse(ReadOnlySpan<char> s, IFormatProvider? provider, out TSelf result);
+    }
+
+    public interface IEquatableOperators<TSelf, TOther> : IEquatable<TOther>
+        where TSelf : IEquatableOperators<TSelf, TOther>
+    {
+        // Given this takes two generic parameters, we could simply call it IEquatable<TSelf, TOther>
+
+        // This should not be `IEquatableOperators<TSelf>` as there are types like `Complex` where
+        // TOther can be `double` and represent an optimization, both in size and perf
+
+        static abstract bool operator ==(TSelf lhs, TOther rhs);
+
+        static abstract bool operator !=(TSelf lhs, TOther rhs);
+    }
+
+    public interface IComparableOperators<TSelf, TOther> : IComparable, IComparable<TOther>, IEquatableOperators<TSelf, TOther>
+        where TSelf : IComparableOperators<TSelf, TOther>
+    {
+        // Given this takes two generic parameters, we could simply call it IComparable<TSelf, TOther>
+
+        // This inherits from IEquatableOperators<TSelf, TOther> as even though IComparable<T> does not
+        // inherit from IEquatable<T>, the <= and >= operators as well as CompareTo itself imply equality
+
+        static abstract bool operator <(TSelf lhs, TOther rhs);
+
+        static abstract bool operator <=(TSelf lhs, TOther rhs);
+
+        static abstract bool operator >(TSelf lhs, TOther rhs);
+
+        static abstract bool operator >=(TSelf lhs, TOther rhs);
+    }
+
+    public interface IAddable<TSelf, TOther, TResult>
+        where TSelf : IAddable<TSelf, TOther, TResult>
+    {
+        // An additive identity is not exposed here as it may be problematic for cases like
+        // DateTime that implements both IAddable<DateTime, DateTime> and IAddable<DateTime, TimeSpan>
+        // This would require overloading by return type and may be problematic for some scenarios where TOther might not have an additive identity
+
+        // We can't assume TOther + TSelf is valid as not everything is commutative
+
+        // We can't expose TSelf - TOther as some types only support one of the operations
+
+        static abstract TResult operator +(TSelf lhs, TOther rhs);
+
+        static abstract TResult checked operator +(TSelf lhs, TOther rhs);
```

Comment:
Yes, there are a number of cases where it would be convenient to use DIMs.

However, I believe DIM support in v1 is still up in the air. We've listed it as a "requirement" for future versioning (so v2+).